weaviate/embedded.py

```diff
@@ -46,11 +49,42 @@ def __init__(self, options: EmbeddedOptions):
         self.ensure_paths_exist()
         self.check_supported_platform()
         self._parsed_weaviate_version = ""
+        # regular expression to detect a version number: v[one digit].[1-2 digits].[1-2 digits] - nothing in front or back
+        pattern = re.compile(r"^v\d\.\d{1,2}\.\d{1,2}$")
```

Comment:
I think the `v` prefix is redundant and can be omitted.

Also, restricting to a suffix of `[1-2 digits] - nothing in front or back` prevents the use of beta or alpha releases.

Reply:
removed the "v", I also added the possibility of adding a "-rc.X" tag. We never had alpha or beta releases for Weaviate, so I left those out

Reply:
See [this comment](https://github.com/weaviate/weaviate-python-client/pull/281#discussion_r1158733405) about alpha releases